synapse/notifier.py

```diff
@@ -274,42 +275,60 @@ def _notify_pending_new_room_events(self, max_room_stream_id: int):
         """
         pending = self.pending_new_room_events
         self.pending_new_room_events = []
+
+        users = set()  # type: Set[Union[str, UserID]]
```

Comment:
Do we expect this to contain both `str` and `UserID`? That could limit the hashing, which is unfortunate.

Reply:
Ah, good point. I mostly put it there to appease mypy, but I'll go and check what is actually going on.